File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/service/oauth/OAuthRequest.java
package com.example.chat.auth.server.core.service.oauth;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotBlank;
import lombok.Builder;

@JsonIgnoreProperties(ignoreUnknown = true)
public record OAuthRequest(
        @NotBlank SocialType provider,
        @NotBlank String code,
        String state,
        String redirectUri) {
    @Builder
    public OAuthRequest {
    }
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/service/oauth/OAuthRequest.kt
package com.example.chat.auth.server.core.service.oauth

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotBlank

@JsonIgnoreProperties(ignoreUnknown = true)
data class OAuthRequest(
        @field:NotBlank val provider: SocialType,
        @field:NotBlank val code: String,
        val state: String? = null,
        val redirectUri: String? = null
)
